Module: komodo_rpc

vaibhavmurkute opened the pull request:
Adds module: komodo_rpc
komodo_rpc.py :
- stores RPC parameters and credentials provided by user, in KomodoRpc class object
- these parameters are read by rpc_util/rpc.py to generate http requests.

* to be merged into dev

vaibhavmurkute committed:
branch refactored

vaibhavmurkute committed:
Merge branch 'dev' into auth

vaibhavmurkute committed:
Adds module: komodo_rpc.py
komodo_rpc.py :
	- stores RPC parameters and credentials provided by user, in KomodoRpc class object
	- these parameters are read by rpc_util/rpc.py to generate http requests.

vaibhavmurkute committed:
added jsonrpc_ver and rpc_req_id in class KomodoRpc so that user can easily modify it, if required.

sjashwin commented on komodo_rpc.py:
```diff
@@ -0,0 +1,29 @@
+class KomodoRpc:
```
Please change `KomodoRpc` to `KomodoRPC`.

vaibhavmurkute replied:
Python PEP-08 style guide suggests, class names should use CapWords (CamelCase) convention. So that's why 'KomodoRpc'. Should I change it to 'KomodoRPC', what do you think?

sjashwin commented on komodo_rpc.py:
```diff
@@ -12,8 +12,10 @@ class KomodoRpc:
+    rpc_req_id = 'curltest'
 
-    def __new__(cls, node_addr='127.0.0.1', rpc_port=7777, req_method='POST', rpc_username='', rpc_password=''):
+    def __new__(cls, node_addr='127.0.0.1', rpc_port=7777, req_method='POST', rpc_username='', rpc_password='', jsonrpc_ver = '1.0', rpc_req_id = 'curltest'):
```
Do not exceed 79 chars a line in python. Break the line down

vaibhavmurkute replied:
It's outdated and has been taken care of, in latest commit.

vaibhavmurkute committed:
- Added docstrings to __new__()
- refactored code to keep characters-per-line count to 79 (PEP-8 Code-Style standard):
	code: 79 chars per line
	docstrings: 72 chars per line

vaibhavmurkute committed:
changed class name KomodoRpc to KomodoRPC
 - changed all other references of this class in other modules
